libs/src/main/scala/retry/retriesimpl/RetryOnAllErrors.scala

```diff
@@ -0,0 +1,36 @@
+package retry.retriesimpl
+
+import cats.MonadError
+import cats.implicits.{catsSyntaxApply, toFlatMapOps, toFunctorOps}
+import retry.retryops.{GiveUp, RetryAndDelay, RetryPolicy}
+import retry.sleep.Sleep
+
+class RetryOnAllErrors[A] {
```

Comment:
it is good but it is too complex comparing to my expectations

generally it is good approach to having sophisticated and extendable solution, but when u need thing like this it is better to use lib directly

try to implement much easier solution with only one method:
```
  def getCustomerFromLegacyService[Out, F[_]: Async](legacyService: F[Out], retries: Int, backoff: FiniteDuration): F[Out] = ...
```